### DISCUSSION TITLE
Exit Code and Signal can be `undefined`

### DISCUSSION THREAD

**Discussion by tsmaeder** [Microsoft/node-pty#753] [Q&A] [answered]
When starting a pty like so:

```
const terminal = spawn(
            command,
            (isWindows && options.commandLine) || options.args || [],
            options.options || {}
        );

        terminal.onExit(({ exitCode, signal }) => {
            console.log(`terminal:exitCode = ${exitCode}, signal =${signal}`);
```
and then later doing `terminal.kill`, I sometimes get `terminal:exitCode = undefined, signal =undefined` in the log. I belies this should never happen, right?

The problem appears on nodejs 20.18.1 on a windows-2019 github action runner. Locally, it seems to always work.

**Comment by tsmaeder** [comment]
Updating the runner image to 2022 fixes the problem. I guess the old image is using winPty. Windows build number is 17763, I believe.  (https://github.com/actions/runner-images/blob/main/images/windows/Windows2019-Readme.md)

**Reply by Tyriar** [reply, marked as answer]
17763 is really old. Yes this means you're stuck on winpty which is legacy and we're not touching it anymore. I recommend updating Windows, but you should be able to use `useConptyDll` in the latest beta builds to get on the newer backend which should be vastly better than winpty (despite being experimental).

https://github.com/microsoft/node-pty/blob/d32b574ccf31f71a2d45694ebcd333433050cb00/typings/node-pty.d.ts#L98-L104
